libcst/codemod/visitors/tests/test_apply_type_annotations.py

```diff
@@ -1119,6 +1119,33 @@ def test_annotate_functions_with_existing_annotations(
             overwrite_existing_annotations=True,
         )
 
+    @data_provider(
+        {
+            "fully_annotated_with_different_stub": (
+                """
+                def f(a: int | str, b: int | list[int | list[int | str]]) -> str: ...
+                """,
+                """
+                def f(a, b):
+                    return 'hello'
+                """,
+                """
+                def f(a: int | str, b: int | list[int | list[int | str]]) -> str:
+                    return 'hello'
+                """,
+            ),
+        }
+    )
+    def test_annotate_functions_pep_604(
+        self, stub: str, before: str, after: str
+    ) -> None:
+        self.run_test_case_with_flags(
+            stub=stub,
+            before=before,
+            after=after,
+            overwrite_existing_annotations=True,
+        )
+
```

Comment:
It might be worth adding a test showing the import collector works (that's where a lot of the new logic lives, and I don't think we have anything exercising that code yet)